app/assets/javascripts/discourse/components/ignored-user-list.js.es6

```diff
@@ -0,0 +1,29 @@
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import showModal from "discourse/lib/show-modal";
+import User from "discourse/models/user";
+
+export default Ember.Component.extend({
+  item: null,
+  actions: {
+    removeItem(item) {
+      this.set("saved", false);
+      this.get("items").removeObject(item);
+      User.findByUsername(item).then(user => {
+        user
+          .updateNotificationLevel("normal")
+          .catch(popupAjaxError)
+          .finally(() => this.set("saved", true));
+      });
+    },
+    newItem() {
+      const modal = showModal("ignore-duration-with-username", {
+        model: this.get("model")
+      });
+      modal.setProperties({
+        onSuccess: username => {
```

Comment:
Why isn't this an action defined in the `ignore-duration-with-username` controller?

Reply:
The `ignore-duration-with-username` controller / modal doesn't have access to `items`. The idea here is that the controller doesn't need to be aware of `items` since its responsibility is to `take care of the ignore user action` and on its success `we` (the `ignored-user-list`) are responsible of what to do next.